checkout_session.go

```diff
@@ -62,6 +62,7 @@ type CheckoutSessionPaymentIntentDataParams struct {
 	ApplicationFeeAmount      *int64                                              `form:"application_fee_amount"`
 	CaptureMethod             *string                                             `form:"capture_method"`
 	Description               *string                                             `form:"description"`
+	Metadata                  map[string]string                                   `form:"metadata"`
```

Comment:
I opted for `map[string]string` for consistency with plan and price. In theory it should be a pointer I think but in this case it's not updatable so it shouldn't matter right?

Reply:
Maps are kind of a special type in that they're always reference types, and so you shouldn't use a pointer with them. See [this blog post on them](https://blog.golang.org/maps):

> ``` go
> var m map[string]int
> ```
>
> Map types are reference types, like pointers or slices, and so the value of m above is nil; it doesn't point to an initialized map. A nil map behaves like an empty map when reading, but attempts to write to a nil map will cause a runtime panic; don't do that. To initialize a map, use the built in make function:
>
> ```
> m = make(map[string]int)
> ```

Reply:
Ah nice, so we didn't overlook this originally when we used map. Thanks for clarifying @brandur-stripe !